arangod/RocksDBEngine/RocksDBHotBackup.cpp

```diff
@@ -156,7 +163,37 @@ std::string RocksDBHotBackup::loadAgencyJson(std::string filename) {
 #else
   return basics::FileUtils::slurp(filename);
 #endif
+}
+
+ResultT<BackupMeta> RocksDBHotBackup::getMeta(std::string const& id) {
+  try {
+    std::string directory = rebuildPath(id);
+
+    std::string metaString =
+      basics::FileUtils::slurp(
+        directory + TRI_DIR_SEPARATOR_CHAR + "META");
+
+    std::shared_ptr<VPackBuilder> metaBuilder =
+      arangodb::velocypack::Parser::fromJson(metaString);
+
+    return BackupMeta::fromSlice(metaBuilder->slice());
 
+  } catch (std::exception const& e) {
+    return ResultT<BackupMeta>::error(TRI_ERROR_HOT_BACKUP_INTERNAL, e.what());
+  }
+}
+
+Result RocksDBHotBackup::writeMeta(std::string const& id, BackupMeta const& meta) try {
```

Comment:
meta contains _id, which should match id parameter.  How about using meta._id instead of passing id parameter.  I doubt there will ever be a bug, but might as well use single data source.

Reply:
This changed over time. At first `_id` was not part of meta but was added later. I think we should separated data that is written and the filename _filename_ which just happens to be also included in the data.